From 91140400d2e4091fe9ca3d0fc60b16a444cffe7f Mon Sep 17 00:00:00 2001
From: Nicolas Le Bayon <nicolas.le.bayon@foss.st.com>
Date: Tue, 26 Nov 2024 17:12:46 +0100
Subject: [PATCH 745/954] docs(dt-bindings): add RNG customization optional
 properties

Add optional properties to set RNG configurations (NIST and health test)
for customization.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git db638d8a5a9f1dd295f4a83df1025d6ea5c64635 docs(dt-bindings): add RNG customization optional properties]
Signed-off-by: Nicolas Le Bayon <nicolas.le.bayon@foss.st.com>
Change-Id: I5a125a73c25d1de9a816513910a91c13d5e9948c
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/421389
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Maxime MERE <maxime.mere@foss.st.com>
Tested-by: Nicolas LE BAYON <nicolas.le.bayon@st.com>
Reviewed-by: Nicolas LE BAYON <nicolas.le.bayon@st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
---
 docs/devicetree/bindings/rng/st,stm32-rng.txt | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/docs/devicetree/bindings/rng/st,stm32-rng.txt b/docs/devicetree/bindings/rng/st,stm32-rng.txt
index 3c613d7914..6cbec44936 100644
--- a/docs/devicetree/bindings/rng/st,stm32-rng.txt
+++ b/docs/devicetree/bindings/rng/st,stm32-rng.txt
@@ -13,6 +13,13 @@ Required properties:
 Optional properties:
 - resets : The reset to properly start RNG
 - clock-error-detect : Enable the clock detection management
+- st,rng-cfg: The configuration bitfields to initialize the RNG. They must
+  be listed in the numerical order. If not specified, the default hard-coded
+  values will be used. Number of values depends on the presence of
+  st,rng-nist-custom property.
+- st,rng-cfg-nist-custom: Enable custom values for NIST compliant RNG.
+- st,rng-htcfg: The values to configure health tests. If not specified, the
+  default hard-coded values will be used.
 
 Example:
 
-- 
2.53.0

